scripts/collect_ad_boilerplate.py

```diff
@@ -0,0 +1,200 @@
+
+    Examples:
+
+    EnableCallbacks -> enable_callbacks
+    EVTLoadGainFile -> e_v_t_load_gain_file
+    """
+
+    # Extend from FileBase class as well if necessary
+    file_base = ''
+    if include_file_base:
+        file_base = ', FileBase'
+
+    # Write class boilerplate
+    boilerplate_file.write(
+        f'''
+class {cam_name}(CamBase{file_base}):
+    _html_docs = ['{dev_name}Doc.html']
+    _default_configuration_attrs = (
+        CamBase._default_configuration_attrs
+    )
+''')
+
+    # Write appropriate attribute for
+    for pv in pv_to_signal_mapping.keys():
+        pv_name = pv
+        if pv_name.endswith('_RBV'):
+            pv_name = pv[:-4]
+        pv_var_name = re.sub('(?<!^)(?=[A-Z])', '_', pv_name).lower()
+        boilerplate_file.write(f"    {pv_var_name} = ADCpt({pv_to_signal_mapping[pv]}, '{pv}')\n")
+
+
+def parse_args():
+    """
+    usage: collect_ad_boilerplate.py [-h] [-t TARGET] [-d]
+
+    Utility for creating boilerplate ophyd classes
+
+    optional arguments:
+      -h, --help            show this help message and exit
+      -t TARGET, --target TARGET
+                            Location of locally installed areaDetector driver
+                            folder structure.
+      -d, --debug           Enable debug loogging for the script.
+    """
+
+    parser = argparse.ArgumentParser(description='Utility for creating boilerplate ophyd classes')
+    parser.add_argument(
+        '-t', '--target', help='Location of locally installed areaDetector driver folder structure.')
+    parser.add_argument('-d', '--debug', action='store_true', help='Enable debug loogging for the script.')
+
+    args = vars(parser.parse_args())
+
+    # Set logging level
+    log_level = logging.ERROR
+    if args['debug']:
+        log_level = logging.DEBUG
+
+    if args['target'] is None:
+        return os.path.abspath('.'), log_level
+    else:
+        return args['target'], log_level
+
+
+if __name__ == '__main__':
+
+    print()
+
+    # Check if input is valid
+    driver_dir, log_level = parse_args()
+    if not os.path.exists(driver_dir) or not os.path.isdir(driver_dir):
+        logging.error(f'Input {driver_dir} does not exist or is not a directory!')
+        exit(-1)
+
+    logging.basicConfig(level=log_level)
+
+    # Check if specified target is an areaDetector driver
+    driver_name = os.path.basename(driver_dir)
+    if not driver_name.startswith('AD'):
+        logging.error(
+            f'Specified driver directory {driver_name} could not be identified as an areaDetector driver!')
+        exit(-1)
+
+    # Collect device, detector, and cam names
+    dev_name = driver_name[2:]
+    det_name = f'{dev_name}Detector'
+    cam_name = f'{det_name}Cam'
+    logging.debug(f'Creating boilerplate for {dev_name}, with classes {det_name} and {cam_name}')
+
+    # Create boilerplate temp file
+    boilerplate_file = open(f'{det_name}_boilerplate', 'w')
```

Comment:
Context manager suggestion here as well. Also, a template file name can be saved and reused on the last line.

I thought that a Jinja2 template can be used for that purpose, but I'm 50/50 on that, as the code is not really large.